pykokkos/core/runtime.py

```diff
@@ -256,7 +291,7 @@ def get_arguments(
             else:
                 is_fused: bool = isinstance(entity, list)
                 if is_fused:
-                    kwargs, _ = fuse_workunit_kwargs_and_params(entity, kwargs)
+                    kwargs, _ = fuse_workunit_kwargs_and_params(entity_trees, kwargs)
```

Comment:
In order to simplify the changes made to `run_workunit`, I would get the list of ASTs here from the parser instead of passing them all the way down.